### PR TITLE
feat($index) + improvements 

### PR DESCRIPTION
**Pull request by Faf4a**
Fixed a random console.log that was forgotten.

Deprecated {selectMenuOption:} as it's also deprecated from discord.js, we should starting to use {stringInput:} and so on.

### CODE REVIEW

**Comment by Faf4a** [comment]
Also, @USERSATOSHI how does aoi.js handle the parser select menus? Couldn't find any const of select menus anywhere.

**Comment by USERSATOSHI** [comment]
> Also, @USERSATOSHI how does aoi.js handle the parser select menus? Couldn't find any const of select menus anywhere.
> 
> I'd like to add parsers for roles/channels and mentionables.

it is at same place you added stringInput support

**Comment by Faf4a** [comment]
Should `$createGuild` and `$editClientAboutMe (no idea what this function should be called)` exist in aoi.js? I mean, there is no discord.js method for this, and would require direct access to the api. (only for the about me one)

**Comment by USERSATOSHI** [comment]
> Should `$createGuild` and `$editClientAboutMe (no idea what this function should be called)` exist in aoi.js? I mean, there is no discord.js method for this, and would require direct access to the api. (only for the about me one)

createguild is fine 
but using an api based function is kinda a no go cause we dont know when that api would go down or something

tho keep the createGuild in  a separate pr

**Comment by Faf4a** [comment]
> createguild is fine but using an api based function is kinda a no go cause we dont know when that api would go down or something
> 
> tho keep the createGuild in a separate pr

alright

**Comment by Faf4a** [comment]
The custom function "fix" would need to be reviewed.

The main issue was that it escaped the brackets, that's fixed with RegEx.

However, the double sending in eval could only be solved by removing the code property which is used in eval from aoi.js-type-custom-functions.

**Comment by Faf4a** [comment]
Interesting enough, aoi.js custom functions aren't registered as functions according to the interpreter?

Whenever using custom functions within a command, it'll return the function itself, however the function does appear in the cache and functions array.

No clue what happened there (tested with and without my changes)

**Comment by Faf4a** [comment]
I found the issue, partially.

The interpreter didn't acknowledge the aoi.js custom functions are functions, therefore I'm forcing it to do so now.

Issue is, now it always returns the custom function as well (aoi.js type).

Would love some revision, thanks.
image

**Comment by Faf4a** [comment]
Reverted modifications of the core and so on, functions are still to be reviewed.

**Comment by Faf4a** [comment]
Added $index to return the current loop index.

**Comment by Faf4a** [comment]
@USERSATOSHI ready for review :)

File: src/handler/parsers.js [hunk 1]
```diff
@@ -163,8 +181,53 @@ const ComponentParser = async (msg, d) => {
       const disabled = inside.shift() === "true";
       const options = inside.join(":").trim();
 
-      let optArray = [];
+      const selectMenuTypes = {
+        userInput: 5,
+        roleInput: 6,
+        mentionableInput: 7,
+        channelInput: 8,
+      }; // types accepted in loop
```
Comment (Leref) [inline src/handler/parsers.js]: Remove comment

**Review by Leref — APPROVED** [review]
There Is more than one change, as $index now exist, changed of loop with index. As well emojiCount updated with default guildID based on user/type is now all.
Lastly, interactonReply/FollowUp now have returnID.

**Pull request merged**